tests/console/zypper_info.pm

```diff
@@ -29,29 +39,29 @@ sub run() {
     my $expected_package_name_vim = 'Name *: vim';
     die "Missing package name. Expected: /$expected_package_name_vim/"
       unless $info_output_vim =~ /$expected_package_name_vim/;
+}
 
-    if (sle_version_at_least('12-SP3')) {
-        # use dvd2 as the src-repository
-        zypper_call('ar --disable --type plaindir cd:///?devices=/dev/sr1 repo-source');
-    }
+sub run() {
+    select_console 'root-console';
 
-    if (check_var('DISTRI', 'opensuse') || sle_version_at_least('12-SP3')) {
-        test_dvd_src_as_repo();
+    # preparation of source repositories
+    if (check_var('DISTRI', 'sle')) {
+        if (get_var('FLAVOR') =~ /-Updates$/) {
+            zypper_call(q{mr -e $(zypper -n lr | awk '/-Source/ {print $1}')});
+        }
+        else {
+            zypper_call('ar --type plaindir cd:///?devices=/dev/sr1 repo-source');
+        }
+    }
+    elsif (check_var('DISTRI', 'opensuse')) {
+        zypper_call("mr -e repo-source");
```

Comment:
please add some comments in this block why the different branches should be considered. It makes sense now but I fear it will not be very clear later. So what about

```perl
# preparation of source repository
my $cmd;
if (check_var('DISTRI', 'sle')) {
    # SLE maintenance tests are assumed to be SCC registered
    if (get_var('FLAVOR') =~ /-Updates$/) {
        $cmd = q{mr -e $(zypper -n lr | awk '/-Source/ {print $1}')};
    }
    # use dvd2 as the src-repository
    else {
        $cmd = 'ar --type plaindir cd:///?devices=/dev/sr1 repo-source';
    }
}
elsif (check_var('DISTRI', 'opensuse')) {
    # openSUSE has a disabled online repository by default
    $cmd = 'mr -e repo-source';
}
zypper_call $cmd;
```